File: SeongHyun/20_validParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        true_table = {
            '(':')',
            '{':'}',
            '[':']'
            
        }
        for string in s:
            if string in ['(', '{', '[']:
                stack.append(string)
            else:
                if not stack:
                    return False
                tmp = stack.pop()
                if true_table[tmp] != string:
                    return False
                
        return len(stack)==0
    

File: EunjiShin/LeetCode/20.valid-parentheses.java
package Easy;

import java.util.HashMap;
import java.util.Map;
import java.util.Stack;

/*
 * @lc app=leetcode id=20 lang=java
 *
 * [20] Valid Parentheses
 */

// @lc code=start
class Solution {

    private static final Map<Character, Character> bracketMap = new HashMap<Character, Character>() {
        {
            put('(', ')');
            put('{', '}');
            put('[', ']');
        }
    };

    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<Character>();
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '{' || ch == '[') {
                stack.push(ch);
            } else if (stack.isEmpty()) {
                return false;
            } else {
                if (bracketMap.get(stack.pop()) != ch) {
                    return false;
                }
            }
        }
        return stack.isEmpty();
    }

}
// @lc code=end
